### ISSUE TITLE
[LiveEffect] Error when running on API 19 emulator

### ISSUE THREAD

**Issue by dturner** [google/oboe#1262]
Android version(s): API 19
Android device(s): API 19 emulator with Google APIs
Oboe version: 1.5.1
App name used for testing: LiveEffect

**Short description**
When running the LiveEffect sample an error is given and no sound is recorded or played.

**Steps to reproduce**
- Create an API 19 emulator. 
- [Enable host audio input. ](https://stackoverflow.com/questions/57189364/android-emulator-not-detecting-host-audio-input)
- Run the LiveEffect app, tap Start

**Expected behavior**
- Audio is recorded from the host microphone

**Actual behavior**
- The error "Engine Failed to Open Streams!" is shown. 

**Device**
```
ro.product.brand = generic_x86
ro.product.manufacturer = unknown
ro.product.model = Android SDK built for x86
ro.product.device = generic_x86
ro.product.cpu.abi = x86
ro.build.description = google_sdk_x86-eng 4.4.2 KK 6761301 test-keys
ro.hardware = ranchu
ro.hardware.chipname = 
ro.arch = 
| grep aaudio = 
```

**Additional info**
Debugging it looks like the opening of the playback stream is failing with `ErrorInvalidFormat` because the stream attempts to open in Float format. 

But why?? It shouldn't be specifying a format.

**Comment by dturner** [comment]
It appears that LiveEffect does all its processing in float, which means (in its current form) it can only run on API 23+ because that's when float input streams were supported.

At the very least LiveEffect's minSDK should be changed from 16 to 23. 

A better solution would be to use int16 on 16-22 and float on 23+.

**Comment by philburk** [comment]
It would be a good example of adapting to different formats. But Oboe provides format conversion.

If developers are going to extend LiveEffect with actual effects processing then they should probably do it in Float. Maybe we should just fix #1261, which would also fix this.

**Comment by dturner** [comment]
I've updated LiveEffect to use format conversion. When opening the input stream on the API 19 emulator it no longer errors with `ErrorInvalidFormat`, instead I get `ErrorInternal`: 

```
07-09 15:00:08.612 4242-4242/com.google.oboe.samples.liveeffect E/AudioRecord: Could not get audio input for record source 6
07-09 15:00:08.612 4242-4242/com.google.oboe.samples.liveeffect E/libOpenSLES: android_audioRecorder_realize(0xb8ca88e0) error creating AudioRecord object
07-09 15:00:08.612 4242-4242/com.google.oboe.samples.liveeffect W/libOpenSLES: Leaving Object::Realize (SL_RESULT_CONTENT_UNSUPPORTED)
07-09 15:00:08.612 4242-4242/com.google.oboe.samples.liveeffect E/OboeAudio: Realize recorder object result:SL_RESULT_CONTENT_UNSUPPORTED
07-09 15:00:08.612 4242-4242/com.google.oboe.samples.liveeffect E/AUDIO-APP: Failed to open input stream. Error ErrorInternal
```

My hypothesis is that OpenSL ES isn't configured for input on the API 19 emulator. I won't be filing a bug for this since we're moving away from supporting API <21. Leaving here for anyone else who might run into this.
